manila-cli.py

```diff
@@ -0,0 +1,110 @@
+
+
+def drop_privileges():
+    """
+    Attempt to drop privileges.
+    """
+    def _try_get(f, *args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except KeyError:
+            return None
+
+    def get_user(name):
+        return _try_get(pwd.getpwnam, name)
+
+    def get_group(name):
+        return _try_get(grp.getgrnam, name)
+
+    # Attempt to find a proper user and group
+    user_info = get_user('nobody')
+    group_info = get_group('nogroup')
+    if group_info is None:
+        group_info = get_group('nobody')
+
+    # Drop privileges
+    if user_info and group_info:
+        os.setegid(group_info.gr_gid)
+        os.seteuid(user_info.pw_uid)
+
+
+if __name__ == '__main__':
+    if os.geteuid() != 0:
+        raise RuntimeError("This program requires superuser privileges")
+
+    # Drop any elevated permissions
+    drop_privileges()
+
+    pre_parser = argparse.ArgumentParser(add_help=False)
+    pre_parser.add_argument('--exports', help='Path to exports file',
+                            default='/etc/exports.conf')
+    pre_parser.add_argument('--basedir', help='Directory where manila '
+                            'filesystems reside, and are reexported',
+                            default='/mnt/manila')
+
+    command_parser = argparse.ArgumentParser(parents=[pre_parser],
+                                             description='Manila exports '
+                                             'management')
+    subparsers = command_parser.add_subparsers(help='sub-command help')
+
+    parser_create = subparsers.add_parser('create', help='Prepare an '
+                                          'export without any access grants')
+    parser_create.add_argument('export_name', help='Export point to create')
+
+    parser_grant = subparsers.add_parser('grant', help='Grant access to an '
+                                         'existing filesystem, and reexport '
+                                         'it')
+    parser_grant.add_argument('export_name',
+                              help='Filesystem to grant access to')
+    parser_grant.add_argument('host', help='IP address or network to grant '
+                              'access for')
+    parser_grant.add_argument('options', help='Export options', nargs='*')
+
+    parser_revoke = subparsers.add_parser('revoke', help='Revoke access from '
+                                          'an existing filesystem, and '
+                                          'reexport it')
+    parser_revoke.add_argument('export_name', help='Filesystem to revoke '
+                               'access from')
+    parser_revoke.add_argument('host', help='IP address or network to revoke '
+                               'access for')
+
+    parser_check = subparsers.add_parser('check', help='Check the environment '
+                                         'for required binaries and running '
+                                         'services')
+
```

Comment:
Is there a way to query the version of the tool, so the Manila driver can query this and validate in `check_configuration`?